### ISSUE TITLE
Some mods bypass all inventory events

### ISSUE THREAD

**Issue by mmis1000** [SpongePowered/SpongeForge#2289]
**I am currently running**

- SpongeForge version: 1.12.2-2705-7.1.0-BETA-3270
- Forge version: 14.23.4.2705
- Java version:  java 8
- Operating System: linux

- Plugins/Mods:
[Inventory Tweaks](https://github.com/Inventory-Tweaks/inventory-tweaks/commit/cca9641f0b03bbef306ee2283fe7daaa3d175100)

**Issue Description**

The `Inventory Tweaks` use a server side mod that calling [player.openContainer.slotClick](https://github.com/Inventory-Tweaks/inventory-tweaks/blob/cca9641f0b03bbef306ee2283fe7daaa3d175100/src/main/java/invtweaks/network/packets/ITPacketClick.java#L50) to accelerate sorting.

But for unknown reason, it turns out that this method completely bypassed the `ClickEvent` and `InverntoryChange` event handlers of sponge plugins and edited the inventory, making cancelling the inventory modify not possible at all.

hold space and left click on a item with `Inventory Tweaks` and its server side mod enabled, you will move the item without trigger any sponge events

And this also makes use chest as a custom gui to provide functionalities harder, as other mods may accidentally tear down the whole gui.

Even edit caused by other mods may not be recoverable, At least we should make the changes detectable. Or it will be very annoying.

**Comment by Faithcaio** [comment]
As this is a custom packet its probably not entering the phase for capturing inventory changes.
I remember having problems with some mods (one was Tinkers Construct I think) going into infinite loops in crafting inventories when canceling the result.
@gabizou

**Comment by mmis1000** [comment]
We have tried change the action to be scheduled at server thread. But it seems it still not works.  
Is this event can be triggered only by normal player action? or it is just another bug?  
If this also should not work, is there a way to manually trigger the event from the Inventory Tweaks's side and give sponge a chance to cancel it?

**Comment by phit** [comment]
confirmed holding down space and clicking still bypasses any events, tested with a simple

```java
    @Listener
    public void onEvent(ClickInventoryEvent event) {
        logger.error(event.toString());
    }
```

InventoryTweaks-1.63.jar
spongeforge-1.12.2-2768-7.1.5-RC3513.jar

**Comment by Faithcaio** [comment]
Looks like the latest 1.64-dev builds of InventoryTweaks make our events fire correctly.

Still there might be other mods using custom packets to change inventory state.

**Comment by phit** [comment]
yes, quark for example https://github.com/SpongePowered/SpongeForge/issues/2553
